### PR TITLE
Add ability to bet on multiple options

### PR DESCRIPTION
**Pull request by ghost**
Dependent on #28 .
Added private helper functions to "Bet" for "resolve" method.
Provided rspec for the updated "resolve" method in "Bet".
With this adjustment, users will be able to place bets on multiple options.

### CODE REVIEW

**Comment by ghost** [comment]
I just realized that the "update_debt" method looks through the whole Debt database and not the debts we are creating at the moment, which might ruin the initial idea of binding debts to specific bets. I will try to resolve it in this PR.

File: app/models/bet.rb [hunk 1]
```diff
@@ -27,4 +23,28 @@ def amount_owed(lost_bet, won_bet)
   def bet_total
     user_bets.pluck(:amount_bet).sum
   end
+
+  private
+
+  def resolve_debt(lost_bet, won_bet)
+    existing_debt = Debt.find_by(debtor: lost_bet.user, creditor: won_bet.user)
+    if existing_debt
+      update_debt(existing_debt, lost_bet, won_bet)
```
Comment (qr8r) [inline app/models/bet.rb:32]: I kind of feel like this is out of scope for this PR ...... I really dont want to worry about duplcated creditor -> debtor debts yet

File: app/models/bet.rb [hunk 2]
```diff
@@ -27,4 +23,28 @@ def amount_owed(lost_bet, won_bet)
...
+
+  def create_debt(lost_bet, won_bet)
+    Debt.create(
+      debtor: lost_bet.user,
+      creditor: won_bet.user,
+      amount:  amount_owed(lost_bet, won_bet)
+    )
+  end
+
+  def update_debt(debt, lost_bet, won_bet)
```
Comment (qr8r) [inline app/models/bet.rb:46]: This is called feature creep. Its funtionality that was not specified in any story and should not have been implemented (yet)

**Review by qr8r — CHANGES_REQUESTED** [review]

File: spec/models/bet_option_spec.rb [hunk 3]
```diff
@@ -48,9 +53,14 @@
     end
 
     context 'New England wins' do
-      let(:option1) { BetOption.create(option_text: 'Atlanta wins') }
+      let(:option1) do
+        BetOption.create(option_text: 'Atlanta wins',
+                         bet: bet1)
```
Comment (jacquesporveau) [inline spec/models/bet_option_spec.rb:58]: In your bet_spec.rb file you are one lining a bet option with two parameters. Pick one syntax style and roll with it.

Comment (qr8r) [reply spec/models/bet_option_spec.rb:58]: Yeah, This might be an artifact of the overly strict rubocop

**Comment by qr8r** [comment]
Jenkins test this please

**Comment by qr8r** [comment]
Closing; No story and tests fail
